src/core_plugins/elasticsearch/lib/patch_kibana_index.js

```diff
@@ -0,0 +1,101 @@
+import {
+  getTypes,
+  getRootType,
+  getRootProperties
+} from '../../../server/mappings';
+
+/**
+ *  Checks that the root type in the kibana index has all of the
+ *  root properties specified by the kibanaIndexMappings.
+ *
+ *  @param  {Object} options
+ *  @property {Function} options.log
+ *  @property {string} options.indexName
+ *  @property {Function} options.callCluster
+ *  @property {EsMappingsDsl} options.kibanaIndexMappingsDsl
+ *  @return {Promise<undefined>}
+ */
+export async function patchKibanaIndex(options) {
+  const {
+    log,
+    indexName,
+    callCluster,
+    kibanaIndexMappingsDsl
+  } = options;
+
+  const rootEsType = getRootType(kibanaIndexMappingsDsl);
+  const currentMappingsDsl = await getCurrentMappings(callCluster, indexName, rootEsType);
+  const missingProperties = await getMissingRootProperties(currentMappingsDsl, kibanaIndexMappingsDsl);
```

Comment:
is this expected to pick up nested properties?  ie if i want to add a new index pattern field.